game/src/ecs/systems/CollisionSystem.java

```diff
@@ -9,63 +9,52 @@
 /** System to check for collisions between two entities */
 public class CollisionSystem extends ECS_System {
 
+    private record CSData(Entity e, HitboxComponent hc) {}
+    ;
+
     /** checks if there is a collision between two entities based on their hitbox */
     @Override
     public void update() {
-        for (Entity entity : ECS.entities) {
+        ECS.entities.stream()
+                .flatMap(e -> e.getComponent(HitboxComponent.class).stream())
+                .map(hc -> buildDataObject((HitboxComponent) hc))
+                .forEach(hc -> secondStream(hc));
```

Comment:
@cagix Geht das auch schöner? Ich möchte für jedes Element im Stream einen weiteren Stream erstellen und mit jedem Element über jedes Element im anderen Stream iterieren (verschachtelte forEach-Schleife).

Reply:
@Lena241 Bin mir nicht sicher, dass ich richtig verstanden habe, was Du vorhast :)

Meinst Du sowas:

```java
for (Entity e1 : ECS.entities) {
    for (Entity e2 : ECS.entities) {
        machWasSchlimmesMit(e1,e2);
    }
}
```
???

Reply:
Und wenn ja, was wäre dann `machWasSchlimmesMit()`?

Reply:
Wenn ich mir den alten Code angucke, ja. 
`checkForCollision()` (wird am Ende von `secondStream()` aufgerufen) ist dann dein `machWasSchlimmesMit()`

Reply:
```java
private record HCPaar (HCData d1, HCData d2){}


     ECS.entities.stream()
                .flatMap(e -> ECS.entities.filter(y->y.id > e.id) .map(erstelleHCPaar).stream())
....
                .map(hc -> buildDataObject((HitboxComponent) hc))
                .map(hc -> secondStream(hc))
                .forEach(hcPaar -> checkForCollision(hcPaar));

```

Glaube, aber das passt eher zu dem aktuellen Überarbeiten des Collisionsystems in #198 . Da weitere Funktionalität dazu kommt.

Reply:
Hmmm, hätte gedacht, dass sowas ein häufig vorkommender Fall wäre und es dafür was in der API gibt. Scheint aber nicht so.

Eigentlich müsste ja nur in https://github.com/Programmiermethoden/Dungeon/pull/400#discussion_r1119716245 die `buildDataObject((HitboxComponent) hc)`-Methode intern noch einmal ein `ECS.entities.stream()`  machen und mit dem übergebenen `hc`-Element dann munter Tupel bauen und dass dann als Stream zurückliefern (dann muss aus dem `map` vorn dran auch ein `flatMap` werden).